lib/liquid/template.rb

```diff
@@ -64,11 +62,11 @@ class << self
       attr_writer :taint_mode
 
       def file_system
-        @@file_system
+        Thread.current[:liquid_file_system] ||= BlankFileSystem.new
```

Comment:
Oh right, but `BlankFileSystem.new` is still wrong a multi-threaded program that initialized this to something else.  So this will at least be a breaking change, and might also be awkward for multi-threaded programs, since they will need to initialize this whenever starting a thread that might render liquid.

You said yourself in the PR description that

> Solved it eventually by assigning it per instance level when rendering templates, which seems to be the most proper solution

so why not allow the filesystem to be overridden on a template instance so this isn't a breaking change.